operator/src/main/java/oracle/kubernetes/operator/helpers/JobStepContext.java

```diff
@@ -287,42 +288,55 @@ protected V1PodSpec createPodSpec(TuningParameters tuningParameters) {
       podSpec.addVolumesItem(additionalVolume);
     }
 
-    List<String> configOverrideSecrets = getConfigOverrideSecrets();
-    for (String secretName : configOverrideSecrets) {
-      podSpec.addVolumesItem(
-            new V1Volume()
-                  .name(secretName + "-volume")
-                  .secret(getOverrideSecretVolumeSource(secretName)));
-    }
-    if (getConfigOverrides() != null && getConfigOverrides().length() > 0) {
-      podSpec.addVolumesItem(
-            new V1Volume()
-                  .name(getConfigOverrides() + "-volume")
-                  .configMap(getOverridesVolumeSource(getConfigOverrides())));
-    }
-    // For WDT
-    if (DomainSourceType.FromModel.toString().equals(getDomainHomeSourceType())) {
-      if (getWdtConfigMap() != null && getWdtConfigMap().length() > 0) {
-        podSpec.addVolumesItem(
-            new V1Volume()
-                .name(getWdtConfigMap() + "-volume")
-                .configMap(getWdtConfigMapVolumeSource(getWdtConfigMap())));
-      }
-      podSpec.addVolumesItem(
-          new V1Volume()
-              .name(RUNTIME_ENCRYPTION_SECRET_VOLUME)
-              .secret(getRuntimeEncryptionSecretVolume()));
+    getConfigOverrideSecrets().forEach(secretName -> addConfigOverrideSecretVolume(podSpec, secretName));
+    Optional.ofNullable(getConfigOverrides()).ifPresent(overrides -> addConfigOverrideVolume(podSpec, overrides));
+
+    if (isSourceWdt()) {
+      Optional.ofNullable(getWdtConfigMap()).ifPresent(mapName -> addWdtConfigMapVolume(podSpec, mapName));
+      addWdtSecretVolume(podSpec);
     }
     return podSpec;
   }
 
+  private void addConfigOverrideSecretVolume(V1PodSpec podSpec, String secretName) {
+    podSpec.addVolumesItem(
+          new V1Volume()
+                .name(secretName + "-volume")
+                .secret(getOverrideSecretVolumeSource(secretName)));
+  }
+
+  private void addConfigOverrideVolume(V1PodSpec podSpec, String configOverrides) {
+    podSpec.addVolumesItem(
+          new V1Volume()
+                .name(configOverrides + "-volume")
+                .configMap(getOverridesVolumeSource(configOverrides)));
+  }
+
+  private boolean isSourceWdt() {
```

Comment:
It probaby ok, any reason not name it as isSourceFromModel to more accurately reflect the logic>

Reply:
No - when I refactored, I kept the names. Everything else there was calling in Wdt, so I didn't change it. But you're correct. We're using the name "fromModel" in a number of other places, and consistency would be good. I'll see where else we use that nomenclature.